src/shell/info.c

```diff
@@ -298,6 +341,10 @@ struct shell_info *shell_info_create (flux_shell_t *shell)
     info->shell_size = rcalc_total_nodes (info->rcalc);
     info->shell_rank = info->rankinfo.nodeid;
     info->total_ntasks = rcalc_total_ntasks (info->rcalc);
+    if (shell_info_set_taskmap (info, create_taskmap (info)) < 0) {
+        shell_log_error ("error creating taskmap");
+        goto error;
+    }
```

Comment:
Same here - taskmap is leaked if `set_taskmap()` fails on one of its early exit paths.

Reply:
Will fix, though failure of this function is fatal error...